### PR TITLE
Fixes #17607 - allow disabling of orphaned repo

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by akofink** [comment]
Getting some strange behavior:

image

**Comment by jlsherrill** [comment]
@akofink updated!   Also, i fixed the issue causing ALL products to show up after deleting the manifest, only ones with repos were supposed to be showing up.  Not sure how long that had been there.

**Comment by akofink** [comment]
@jlsherrill It looks much better, but I do have to ask if this is right:

image

Why is RHEL7 Server RPMs 7.0 (enabled for this test, not related to any content views/hosts) showing up under the 6.2 tools repo-set as well as RHEL 7 Server? It appears that all my repos are showing under each repo set instead of just in their own respective sets.

**Comment by jlsherrill** [comment]
@akofink good catch, updated!

File: app/controllers/katello/products_controller.rb [hunk 1]
```diff
@@ -80,8 +79,8 @@ def exclude_rolling_kickstart_repos(repos)
       end
     end
 
-    def available_synced_repos(repos)
-      @product.repositories.each do |product_repo|
+    def available_synced_repos(repos, content_id)
+      @product.repositories.in_default_view.where(:content_id => content_id).each do |product_repo|
```
Comment (akofink) [inline app/controllers/katello/products_controller.rb]: s/each/find_each/ ? Though, I assume a user having a whole bunch of repos (> 1000) for a given product is pretty rare. I'm surprised hound doesn't catch this.

Comment (jlsherrill) [inline app/controllers/katello/products_controller.rb]: Yeah it is pretty rare, but i don't think there's a reason to not use find_each. 👍

Comment (jlsherrill) [inline app/controllers/katello/products_controller.rb]: added and pushed

Comment (akofink) [inline app/controllers/katello/products_controller.rb]: ACK

**Comment by jlsherrill** [comment]
[test]

**Pull request merged**